app/models/models.py

```diff
@@ -39,102 +43,137 @@ def hashed_password(password):
 
     @staticmethod
     def get_user_by_id(user_id):
-        user = User.query.filter_by(id=user_id).first()
+        users = user_container.query_items(
+            query="SELECT * FROM users u WHERE u.id = @userId",
+            parameters=[dict(name="@userId", value=user_id)],
+            enable_cross_partition_query=True,
+        )
+        user = list(users)[0]
+
+        return user
+
+    @staticmethod
+    def get_user_by_email(user_email):
+        users = user_container.query_items(
+            query="SELECT * FROM users u WHERE u.email = @userEmail",
+            parameters=[dict(name="@userEmail", value=user_email)],
+            enable_cross_partition_query=True,
+        )
+        user = list(users)[0]
+
         return user
 
     @staticmethod
     def get_user_with_email_and_password(email, password):
```

Comment:
We should be using OAuth to manage users and permissions.  Having a method with email and password basically means that your app controls user access, permissions, etc and storing of credentials in the database.  Typically you avoid with this access tokens and OAuth.  It's a bigger more complicated implementation that you should learn, but we can discuss later.

Reply:
Yes, I'd love to have a discussion on this. Right now, one problem I'm seeing is that when a user is logged in, he/she can edit other people's tasks, instead of just his/her own. So basically, a logged in user has access to the whole database, instead of just part of it that stores his/her own info. I don't know if this can be solved by OAuth, but the way this works now is not ideal. 

Also, can we discuss the create-react-app too? I didn't change the structure of the code at all. I think the original app is based on this [boilerplate](https://github.com/dternyak/React-Redux-Flask) that combines Flask and React, which I guess is not ideally structured.